Actions/DeployPowerPlatform/action.yaml

```diff
@@ -0,0 +1,153 @@
+name: Deploy PowerPlatform Solution
+author: Microsoft Corporation
+inputs:
+  shell:
+    description: Shell in which you want to run the action (powershell or pwsh)
+    required: false
+    default: powershell
+  actionsRepo:
+    description: The AL-Go for GitHub Actions repository
+    required: true
+  environmentName:
+    description: Name of environment to deploy to
+    required: true
+  artifactsFolder:
+    description: Path to the downloaded artifacts to deploy (when deploying from a build)
+    required: false
+    default: ''
+  solutionFolder:
+    description: Path to the unpacked solution to deploy (when deploying from branch)
+    required: false
+    default: ''
+  deploymentEnvironmentsJson:
+    description: The settings for all Deployment Environments
+    required: true
+runs:
+  using: composite
+  steps:
+    - name: Install Power Platform Tools
+      uses: microsoft/powerplatform-actions/actions-install@v1
+
+    - name: Get Actions Repo and Ref
+      shell: ${{ inputs.shell }}
+      env:
+        _actionsRepo: ${{ inputs.actionsRepo }}
+      run: |
+        # Using hardcoded reference to microsoft/AL-Go-Actions inside composote actions doesn't work with preview nor direct development branches
+        # For this reason, we get the actionsRepo from the calling workflow (which might be freddydk/AL-Go@PPP or microsoft/AL-Go@sha or microsoft/AL-Go-Actions@v4.1 )
+        # Download actions to ./_AL-Go/Actions
+        $actionsRef = $env:_actionsRepo.Split('@')[1]
+        $actionsRepo = $env:_actionsRepo.Split('@')[0]
+        $actionsPath = './_AL-Go/Actions'
+        if ($actionsRepo -like '*/Actions') {
+          $actionsRepo = $actionsRepo.SubString(0,$actionsRepo.Length-8)
+          $actionsPath = './_AL-Go'
+        }
+        Add-Content -encoding utf8 -path $env:GITHUB_ENV -value "actionsRepo=$actionsRepo"
+        Add-Content -encoding utf8 -path $env:GITHUB_ENV -value "actionsRef=$actionsRef"
+        Add-Content -encoding utf8 -path $env:GITHUB_ENV -value "actionsPath=$actionsPath"
+
+    - name: Check out AL-Go Actions
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ env.actionsRepo }}
+        ref: ${{ env.actionsRef }}
+        path: ${{ env.actionsPath }}
+
+    - name: Parse DeployToSettings and AuthContext
+      id: ReadPowerPlatformSettings
+      uses: ./_AL-Go/Actions/ReadPowerPlatformSettings
```

Comment:
Why doesn't it just use the action from microsoft/AL-Go-Actions?

Reply:
The DeployPowerPlatform action is not part of the AL-Go template and as such is not changed to point to the AL-Go actions used during development. Using microsoft/AL-Go-Actions would cause this to always use the latest version of the actions and not the version you are calling or the version in your development branch.
So having an action which invokes other actions in the same version of actions, needs to get the Actions link from the caller (which is then changed by Update AL-Go System Files to point to the version of the actions we are calling)